python/paddle/fluid/tests/unittests/test_multimarginloss.py

```diff
@@ -0,0 +1,348 @@
+# -*- coding: utf-8 -*
+#   Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+import numpy as np
+import unittest
+
+
+def call_MultiMarginLoss_layer(
+    input,
+    label,
+    p=1,
+    margin=1.0,
+    weight=None,
+    reduction='mean',
+):
+    triplet_margin_loss = paddle.nn.MultiMarginLoss(p=p,
+                                                    margin=margin,
+                                                    weight=weight,
+                                                    reduction=reduction)
+    res = triplet_margin_loss(
+        input=input,
+        label=label,
+    )
+    return res
+
+
+def call_MultiMarginLoss_functional(
+    input,
+    label,
+    p=1,
+    margin=1.0,
+    weight=None,
+    reduction='mean',
+):
+    res = paddle.nn.functional.multi_margin_loss(input=input,
+                                                 label=label,
+                                                 p=p,
+                                                 margin=margin,
+                                                 weight=weight,
+                                                 reduction=reduction)
+    return res
+
+
+def test_static(place,
+                input_np,
+                label_np,
+                p=1,
+                margin=1.0,
+                weight_np=None,
+                reduction='mean',
+                functional=False):
+    prog = paddle.static.Program()
+    startup_prog = paddle.static.Program()
+    with paddle.static.program_guard(prog, startup_prog):
+        input = paddle.static.data(name='input',
```

Comment:
For test cases in static mode, you'd better add some cases that `paddle.static.data` creates data layer that has dynamic shape to ensure that it works as expected.

Reply:
I add the new test cases of checking data shape in static mode. but I'm not quite sure about it.

Reply:
I mean, you can create an data layer with shape `[-1, -1]` and feed it with an array with shape `[3, 4]` or `[5, 6]`.
Or create a data layer with shape `[-1, 3,4]` and feed it with an array with shape `[7, 3, 4]`

I the rank and all the static size matches, it is compatible.